src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -1804,7 +1804,7 @@ namespace winrt::TerminalApp::implementation
         try
         {
             auto parsed = winrt::Windows::Foundation::Uri(eventArgs.Uri().c_str());
-            if (parsed.SchemeName() == L"http" || parsed.SchemeName() == L"https")
+            if (_IsUriSupported(parsed))
             {
                 ShellExecute(nullptr, L"open", eventArgs.Uri().c_str(), nullptr, nullptr, SW_SHOWNORMAL);
             }
```

Comment:
Wouldn't it be better to use the "edit" verb here for file URLs? Perhaps having two separate conditions, like `IsUriSupportedForOpen`, and `IsUriSupportedForEdit`. Because if a user is clicking on a link to a `.py` file, I think it's more likely they would want the file edited rather than executed (which is what the _open_ verb tends to do).

I don't know if there are any downsides to using the _edit_ verb though. I did bring this up in the #7562 discussion thread, but I don't think anyone commented on the merits of the idea one way or the other.

Reply:
I'm not in love with the idea of using the edit verb. I just gave it a test on a couple of the files I had laying around:

extension|open behavior|edit behavior
-|-|-
ps1|notepad|powershell ISE (?!)
txt|notepad|notepad
cpp|visual studio preview, which is not my default (argh)|notepad
_no extension_|"how do you want to open this?" dialog|an error message from ShellExecute
.gitconfig, .wslconifg, etc|same as above|same as above

It's great if somebody's registered for it, but I'm bothered that the OS doesn't give it the same "do what with?" treatment.

Reply:
The extensions that concerned me were things like `.py`, `.cmd`, `.bat`, and `.vbs`. In all cases, the _open_ verb executed the script, and the `edit` verb opened the file in notepad. Those all seem like files that would _not_ be good to execute when you click on them.

If we're worried about the cases where a file extension doesn't have a registered handler, maybe we can handle that as a special case check. But wanting to click on a link to a file that you've never registered a handler for, seems a far less likely use case than clicking on something like a `.py` file, and expecting to be able to edit it.

Although maybe I'm the odd one out here. I just assumed nobody would want scripts to execute when clicking on their links.